### PR TITLE
:sparkles: Refactor HUB API Jira changes

### PR DESCRIPTION
**Pull request by gildub**
Resolves https://github.com/konveyor/tackle2-ui/issues/1053

### CODE REVIEW

File: client/src/app/pages/migration-waves/components/status-table.tsx [hunk 1]
```diff
@@ -84,18 +84,17 @@ export const WaveStatusTable: React.FC<IWaveStatusTableProps> = ({
...
     if (appId) {
       const ticket = getTicketByApplication(tickets, appId);
       if (ticket) {
-        const types = getTrackerTypesByProjectId(
-          trackers,
+        const types = useTrackerTypesByProjectId(
           ticket.tracker.name,
-          ticket?.parent
+          ticket.parent
         );
```
Comment (mturley) [inline client/src/app/pages/migration-waves/components/status-table.tsx:98]: This has the same issue we identified before with `useTrackerProjectsByTracker` where it is conditionally calling a hook. It should be renamed `useTicketIssue`, and the `useTrackerTypesByProjectId` should be called outside the `if (ticket)` block, which will require its params to be optional (internally passing `enabled: false` to useQuery if they are missing). Then make sure `useTicketIssue` is not being called in a conditional or callback (called at render time, every render).

**Review by mturley — CHANGES_REQUESTED** [review]
One more hook rules issue here, otherwise LGTM.

File: client/src/app/pages/migration-waves/components/status-table.tsx (hunk 1 above)
Comment (gildub) [reply client/src/app/pages/migration-waves/components/status-table.tsx]: Thanks @mturley , I removed the hook from being under condition.

**Review by mturley — APPROVED** [review]
LGTM

**Pull request merged**